src/connectivity/components.c

```diff
@@ -1246,6 +1250,167 @@ igraph_error_t igraph_biconnected_components(const igraph_t *graph,
+
+    for (igraph_integer_t i = 0; i < no_of_nodes; i++) {
+
+        if (VECTOR(low)[i] != 0) {
+            continue;    /* already visited */
+        }
+
+        IGRAPH_ALLOW_INTERRUPTION();
+
+        IGRAPH_CHECK(igraph_stack_int_push(&path, i));
+        counter = 1;
+        rootdfs = 0;
+        VECTOR(low)[i] = VECTOR(num)[i] = counter++;
+        while (!igraph_stack_int_empty(&path)) {
+            igraph_integer_t n;
+            igraph_integer_t act = igraph_stack_int_top(&path);
+            igraph_integer_t actnext = VECTOR(nextptr)[act];
+
+            neis = igraph_lazy_adjlist_get(&inclist, act);
+            n = igraph_vector_int_size(neis);
+            if (actnext < n) {
+                /* Step down (maybe) */
+                igraph_integer_t nei = VECTOR(*neis)[actnext];
+                if (VECTOR(low)[nei] == 0) {
+                    if (act == i) {
+                        rootdfs++;
+                    }
+                    IGRAPH_CHECK(igraph_stack_int_push(&path, nei));
+                    VECTOR(low)[nei] = VECTOR(num)[nei] = counter++;
+                } else {
+                    /* Update low value if needed */
+                    if (VECTOR(num)[nei] < VECTOR(low)[act]) {
+                        VECTOR(low)[act] = VECTOR(num)[nei];
+                    }
+                }
+                VECTOR(nextptr)[act] += 1;
+            } else {
+                /* Step up */
+                igraph_stack_int_pop(&path);
+                if (!igraph_stack_int_empty(&path)) {
+                    igraph_integer_t prev = igraph_stack_int_top(&path);
+                    /* Update LOW value if needed */
+                    if (VECTOR(low)[act] < VECTOR(low)[prev]) {
+                        VECTOR(low)[prev] = VECTOR(low)[act];
+                    }
+                    /* Check for articulation point */
+                    if (VECTOR(low)[act] >= VECTOR(num)[prev]) {
+                        if (prev != i /* the root */) {
+                            /* Found an articulation point so the graph is not
+                             * biconnected */
+                            is_biconnected = false;
+                            goto exit;
+                        }
+                    }
+                } /* !igraph_stack_int_empty(&path) */
+            }
+
+        } /* !igraph_stack_int_empty(&path) */
+
+        if (rootdfs >= 2) {
+            is_biconnected = false;
+            goto exit;
+        }
+
+    } /* i < no_of_nodes */
+
+exit:
+
+    igraph_lazy_adjlist_destroy(&inclist);
+    igraph_stack_int_destroy(&path);
+    igraph_vector_int_destroy(&low);
+    igraph_vector_int_destroy(&num);
+    igraph_vector_int_destroy(&nextptr);
+    IGRAPH_FINALLY_CLEAN(5);
+
+exit2:
+
+    if (res) {
+        *res = is_biconnected;
+    }
+
+    /* We do not touch the cache for graphs with less than three nodes because
+     * of all the edge cases. */
+    if (is_biconnected && no_of_nodes > 2) {
```

Comment:
```suggestion
    if (is_biconnected && no_of_nodes >= 3) {
```

Just for consistency with the lines at the top that read the cache.